### DISCUSSION TITLE
Ability to validate email for a user from the organization page

### DISCUSSION THREAD

**Discussion by Yberion** [zitadel/zitadel#7403] [Ideas]
Hello,

It could be good to validate the email of a user made with the register page. Or it's more about ability to Activate the user?

Currently we can only `Resend Code`:

![image](https://github.com/zitadel/zitadel/assets/4186385/23688850-c3fb-4420-b36f-5a418367c1fd)

It would be good to have the same option as when we create a new user from the organization page:

![image](https://github.com/zitadel/zitadel/assets/4186385/635d2e17-8085-4e50-bdc3-ebd3bed629d6)

**Comment by hifabienne** [comment]
Hei @Yberion 
I don't really understand what you are looking for.
When a user is created and the email is not marked as verified, the user will get the verification email. 
When clicking on that link the email is marked as verified. 
This should happen no matter where the user is created

**Reply by Yberion** [reply]
Hello @hifabienne,

Sorry if I wasn't clear.

When an user create himself an account there is no way to tick the "Email Verified" checkbox as it is not available, probably because the user is not "Active".

![image](https://github.com/zitadel/zitadel/assets/4186385/c4379ce3-2fc9-4d52-9b86-3e39045c7347)

When I create myself an account for a user on the UI Console, the user is directly marked as "Active" and I suppose it give us the ability to I have the "Email Verified" checkbox in the UI Console.

![image](https://github.com/zitadel/zitadel/assets/4186385/a5aed7bc-e9ea-4097-9213-4066d15c57ad)

Is there a way to active an user or having the "Email Verified" checkbox like when I create myself an user?

**Reply by hifabienne** [reply]
When you create a new user in the console you have the possibility to check the email is verified box and set an initial password. If you do so, the user will not receive an initialization email and is active directly:
<img width="979" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/b302c9a6-8618-41ac-9585-3e778441658f">

If the users register themself trhough the register page, there is no option for them to check that box. As this would not make sense, because the user could skip the verifcation process in that case.
I am not 100% sure, but I think you could manipulate the user trhough an action and set the email to verified if thats what you want to achieve. 
<img width="979" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/1556cc6c-6dea-4a68-af7e-f8e40d838a9e">

**Reply by Yberion** [reply]
Hello @hifabienne,

Thanks for the answer, this is really appreciated to get that fast feedback 🙏 

> As this would not make sense, because the user could skip the verifcation process in that case.

Well, we can already skip the verification by creating ourself the user instead. That would make sens to me to be able to skip the verification when a user create himself as well, but I understand that it's not possible.

At work we're using Keycloak and this is possible on it, that's why I asked for the idea there, there's sometime a use for this.

![image](https://github.com/zitadel/zitadel/assets/4186385/75697d22-2099-47eb-a8f9-dd5730259396)

![image](https://github.com/zitadel/zitadel/assets/4186385/fb5915c8-203f-4c3b-acc3-2f5b9c4214c4)

And if I toggle the button the user is now verified:

![image](https://github.com/zitadel/zitadel/assets/4186385/8eeb7677-80f6-4ee5-8cce-ca0882c5c5f0)

I can close the discussion if it makes no sens on Zitadel or create a feature request.

---

> I am not 100% sure, but I think you could manipulate the user trhough an action and set the email to verified if thats what you want to achieve.

Yeah it seems to be possible with https://zitadel.com/docs/apis/actions/internal-authentication#parameters-of-pre-creation, there's also an example there https://github.com/zitadel/actions/blob/main/examples/verify_email.js

But I can't make it to work, I'm probably doing something wrong:

![image](https://github.com/zitadel/zitadel/assets/4186385/af5ad56f-d678-44b0-a3af-407b4d8a6073)

I can make a new discussion (or bug?) if needed regarding my issue

**Reply by hifabienne** [reply]
the name of the action should match the function name in the java script
could your try changing that?

**Reply by Yberion** [reply]
Hello @hifabienne, sorry for the late answer.

I've just tested to set the same name as the function and it does indeed work.

My apologies, I should've read more carefully the doc!

The main point still remains, it would be good to have this checkbox even for manually created users.

**Reply by doraeric** [reply]
How to call that API? Can that be done in browser dev tools?
I found a checkbox on the page, but clicking it doesn't change anything.
![image](https://github.com/zitadel/zitadel/assets/16789570/d4043bc3-a3f5-41a7-a200-ef20733d0fd6)

**Reply by hifabienne** [reply]
hei, I quickly tested this and you are right.
There is already a bug report to this: https://github.com/zitadel/zitadel/issues/7762

**Comment by tyteen4a03** [comment]
It would be great if this feature is enabled for admins in the console, regardless of user status.
